File: Tunisian-Camp/Team/src/main/java/Esprit/CampProject/Team/entities/Localisation.java
package Esprit.CampProject.Team.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;
import javax.persistence.*;
import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

/**
 * A Localisation.
 */
@Entity

public class Localisation implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "adress")
    private String adress;

    @OneToMany(mappedBy = "localisation")
    @Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
    @JsonIgnoreProperties(value = { "localisation", "forum", "reservations", "activities" }, allowSetters = true)
    private Set<Camping> campings = new HashSet<>();

    // jhipster-needle-entity-add-field - JHipster will add fields here

    public Long getId() {
        return this.id;
    }

    public Localisation id(Long id) {
        this.setId(id);
        return this;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public Localisation name(String name) {
        this.setName(name);
        return this;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAdress() {
        return this.adress;
    }

    public Localisation adress(String adress) {
        this.setAdress(adress);
        return this;
    }

    public void setAdress(String adress) {
        this.adress = adress;
    }

    public Set<Camping> getCampings() {
        return this.campings;
    }

    public void setCampings(Set<Camping> campings) {
        if (this.campings != null) {
            this.campings.forEach(i -> i.setLocalisation(null));
        }
        if (campings != null) {
            campings.forEach(i -> i.setLocalisation(this));
        }
        this.campings = campings;
    }

    public Localisation campings(Set<Camping> campings) {
        this.setCampings(campings);
        return this;
    }

    public Localisation addCamping(Camping camping) {
        this.campings.add(camping);
        camping.setLocalisation(this);
        return this;
    }

    public Localisation removeCamping(Camping camping) {
        this.campings.remove(camping);
        camping.setLocalisation(null);
        return this;
    }

    // jhipster-needle-entity-add-getters-setters - JHipster will add getters and setters here

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Localisation)) {
            return false;
        }
        return id != null && id.equals(((Localisation) o).id);
    }

    @Override
    public int hashCode() {
        // see https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/
        return getClass().hashCode();
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "Localisation{" +
            "id=" + getId() +
            ", name='" + getName() + "'" +
            ", adress='" + getAdress() + "'" +
            "}";
    }
}

File: t_campy_front/src/app/Models/Localisation/localisation.ts
import { Camping } from '../Camping/camping';

export class Localisation {
  private id: number;
  private name: string;
  private address: string;
  private campings: Camping[];

  constructor(id: number, name: string, address: string, campings: Camping[]) {
    this.id = id;
    this.name = name;
    this.address = address;
    this.campings = campings;
  }

  public static fromJson(json: any): Localisation {
    if (json === null) {
      return null as any;
    }
    return new Localisation(
      json.id,
      json.name,
      json.address,
      Camping.fromJsonArray(json.campings)
    );
  }

  public static fromJsonArray(jsonLocalisation: any[]): Localisation[] {
    const localisations: Localisation[] = [];
    jsonLocalisation.forEach((element) => {
      localisations.push(Localisation.fromJson(element));
    });
    return localisations;
  }

  public toJson(): any {
    return {
      id: this.id,
      name: this.name,
      address: this.address,
      campings: this.campings.map((camping) => camping.toJson()),
    };
  }

  public getId(): number {
    return this.id;
  }

  public getName(): string {
    return this.name;
  }

  public getAddress(): string {
    return this.address;
  }

  public getCampings(): Camping[] {
    return this.campings;
  }

  public setId(id: number): void {
    this.id = id;
  }

  public setName(name: string): void {
    this.name = name;
  }

  public setAddress(address: string): void {
    this.address = address;
  }

  public setCampings(campings: Camping[]): void {
    this.campings = campings;
  }

  public static empty(): Localisation {
    return new Localisation(0, '', '', []);
  }

  public toString(): string {
    return this.name + ' ' + this.address;
  }

  public equals(localisation: Localisation): boolean {
    return this.id === localisation.getId();
  }

  public compareTo(localisation: Localisation): number {
    if (this.id > localisation.getId()) {
      return 1;
    } else if (this.id < localisation.getId()) {
      return -1;
    } else {
      return 0;
    }
  }
}
